lib/cli.js

```diff
@@ -382,11 +382,22 @@ export default async function loadCli() { // eslint-disable-line complexity
 	}
 
 	let parallelRuns = null;
+	let parallelRunsComparator = null;
 	if (isCi && ciParallelVars) {
 		const {index: currentIndex, total: totalRuns} = ciParallelVars;
 		parallelRuns = {currentIndex, totalRuns};
 	}
 
+	if (parallelRuns) {
+		if (Reflect.has(conf, 'ciParallelRunsComparator') && typeof conf.ciParallelRunsComparator !== 'function') {
```

Comment:
We could make this more generic still, say `sortTestFiles`. Would let you randomize execution order as well (or, if we decide to randomize, it provides a way for others to do a deterministic sort).

Reply:
> Do you mean generic for all runs, or only CI runs?
> With caching and running failed tests first, a sorter might be confusing for local runs.
> Happy to implement it, if that's the request.

For all runs, yea. I'm assuming the failure prioritization keeps the relative order of the prioritized files, I don't think that'll be too confusing. And otherwise you could detect CI in the configuration and act accordingly.

> Also, can you confirm my understanding that currently tests files are ordered based on the return value of `globby`?

I suppose, can't quite recall if there's multiple concatenations or whatnot. But the current order is not defined. I suppose that means we shouldn't set a default comparator and only sort when one is provided.

Reply:
Done in https://github.com/avajs/ava/pull/2968/commits/9836cd7ae8b6083a2c0b4a2dc289809d5b10f6dd!

Going to add a separate test for non CI runs

Reply:
Added another test in https://github.com/avajs/ava/pull/2968/commits/44cdb863c1efdde52e4fec3fcc62fc1c6dced5b2.
Please let me know if I put it in the right place.

I used `concurrency: 1` to make the test output predictable.
With AVA parallelism I'm not sure sorting can guarantee order of execution otherwise, WDYT?